File: typescript/src/domain/referral/wire.ts
export interface ReferralCodeWire {
  code: string;
  max_uses: number;
  use_count: number;
}

export interface ReferralStatusResponse {
  is_beta: boolean;
  source?: string;
  referral_codes: ReferralCodeWire[];
}

export interface RedeemRequest {
  code: string;
}

export interface RedeemResponse {
  success: boolean;
  is_beta: boolean;
}

export interface RedeemErrorResponse {
  error: string;
  code: string;
}

File: rust/src/domain/referral/wire.rs
//! Wire types for referral API responses.

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReferralStatusResponse {
    pub is_beta: bool,
    pub source: Option<String>,
    pub referral_codes: Vec<ReferralCodeWire>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReferralCodeWire {
    pub code: String,
    pub max_uses: i32,
    pub use_count: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RedeemRequest {
    pub code: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RedeemResponse {
    pub success: bool,
    pub is_beta: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RedeemErrorResponse {
    pub error: String,
    pub code: String,
}
